propnet/core/graph.py

```diff
@@ -242,11 +272,27 @@ def get_types(symbols_in, legend):
                     to_return.append(out)
                 return to_return
 
+            # list<list<SymbolType>>, representing sets of input properties the model accepts.
             type_inputs = get_types(sym_inputs, legend)
 
             # Look through all input sets and match with all combinations from lookup_dict.
             def gen_input_dicts(symbols, candidate_props, constraint_props, level):
-                """Recursively generates all possible combinations of input arguments"""
+                """
+                Recursively generates all possible combinations of input arguments.
+                Args:
+                    symbols (list<str>): one set of input symbols required by the model.
+                    candidate_props (list<list<Symbol>>):
+                        list of potential values that can be plugged into each symbol,
+                        the outer list corresponds by ordering to the symbols list,
+                        the inner list gives values that can be plugged in to each symbol.
+                    constraint_props (dict<str, lambda(Symbol) -> bool>):
```

Comment:
Genuine question, is this correct syntax for a type annotation like this? I'm not sure what convention google docstrings use. By strict Python PEP 484 this is wrong, but I'm not sure it's worth the time getting the syntax strictly right. It's clear what you mean and we've been quite lax with type hints elsewhere...

Reply:
If I had to guess, it very likely is not correct. I'll try to look up how these things are documented by others. I just kind of followed how it's done in Swift.

At least it is clear what I mean!